libsigner/src/messages.rs

```diff
@@ -1039,6 +1116,87 @@ impl From<BlockValidateReject> for SignerMessage {
     }
 }
 
+/// Signed network packets sent by signers along with their view
+/// of stacks tip consensus hash and block height
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct PacketMessage {
+    /// Signed network packet
+    pub packet: Packet,
+    /// The signer's id
+    pub packet_signer_id: u32,
+    /// The signer's view of coordinator metadata including stacks tip consensus hash and block height
+    pub coordinator_metadata: CoordinatorMetadata,
+}
+
+impl PacketMessage {
+    /// Create a new PacketMessage
+    pub fn new(
+        packet: Packet,
+        packet_signer_id: u32,
+        coordinator_metadata: CoordinatorMetadata,
+    ) -> Self {
+        PacketMessage {
+            packet,
+            packet_signer_id,
+            coordinator_metadata,
+        }
+    }
+}
+
+/// Calculated Coordinator Metadata per signer's view of Stacks chain
+#[derive(Clone, Copy, Debug, Hash, Eq, PartialEq, Serialize, Deserialize)]
+pub struct CoordinatorMetadata {
+    /// The stacks_tip_consensus_hash from the signer's view
```

Comment:
You should be using `pox_consensus` here, since it's the value that advances with bitcoin blocks. `stacks_tip_consensus_hash` advances with Stacks blocks.  If there's a chain stall, you'll want to use a consensus hash value that advances without the Stacks nodes' help.